certbot/ocsp.py

```diff
@@ -62,12 +79,35 @@ def ocsp_revoked(self, cert_path, chain_path):
         logger.debug(" ".join(cmd))
         try:
             output, err = util.run_script(cmd, log=logger.debug)
-        except errors.SubprocessError:
+        except errors.SubprocessError:  # pragma: no cover
             logger.info("OCSP check failed for %s (are we offline?)", cert_path)
             return False
-
         return _translate_ocsp_query(cert_path, output, err)
 
+    @staticmethod
+    def _ocsp_revoke_cryptography(cert_path, chain_path, url):
+        with open(chain_path, 'rb') as file_handler:
+            issuer = x509.load_pem_x509_certificate(file_handler.read(), default_backend())
+        with open(cert_path, 'rb') as file_handler:
+            cert = x509.load_pem_x509_certificate(file_handler.read(), default_backend())
+        builder = x509.ocsp.OCSPRequestBuilder()
+        builder = builder.add_certificate(cert, issuer, SHA1())
+        request = builder.build()
+        request_binary = request.public_bytes(serialization.Encoding.DER)
+        response = requests.post(url, data=request_binary,
+                                 headers={'Content-Type': 'application/ocsp-request'})
+        if not response.status_code == 200:
+            logger.info("OCSP check failed for %s (are we offline?)", cert_path)
+            return False
+        response_ocsp = ocsp.load_der_ocsp_response(response.content)
+        try:
+            logger.debug("OCSP certificate status for %s is: %s",
+                         cert_path, response_ocsp.certificate_status)
+            return response_ocsp.certificate_status == ocsp.OCSPCertStatus.REVOKED
```

Comment:
This is not actually checking the signature of the OCSP response against the issuer's key, so this is vulnerable to MITM attacks.

Reply:
Indeed, I did not catch it, thanks ! It is corrected: @alex, is it ok for you ?